kvdb/sqlbase/schema.go

```diff
@@ -0,0 +1,70 @@
+//go:build kvdb_postgres
+
+package sqlbase
+
+import (
+	"fmt"
+	"strings"
+)
+
+// SQLiteCmdReplacements is a one to one mapping of sqlite keywords that should
+// be replaced by the mapped strings in any command.
+type SQLiteCmdReplacements map[string]string
+
+func newKVSchemaCreationCmd(table, schema string,
+	replacements SQLiteCmdReplacements) string {
+
+	var (
+		tableInSchema = table
+		finalCmd      string
+	)
+	if schema != "" {
+		finalCmd = fmt.Sprintf(
+			`CREATE SCHEMA IF NOT EXISTS ` + schema + `;`,
+		)
+
+		tableInSchema = fmt.Sprintf("%s.%s", schema, table)
+	}
+
+	// Construct the sql statements to set up a kv table in postgres. Every
+	// row points to the bucket that it is one via its parent_id field. A
+	// NULL parent_id means that the key belongs to the upper-most bucket in
+	// this table. A constraint on parent_id is enforcing referential
+	// integrity.
+	//
+	// Furthermore there is a <table>_p index on parent_id that is required
+	// for the foreign key constraint.
+	//
+	// Finally there are unique indices on (parent_id, key) to prevent the
+	// same key being present in a bucket more than once (<table>_up and
+	// <table>_unp). In postgres, a single index wouldn't enforce the unique
+	// constraint on rows with a NULL parent_id. Therefore two indices are
+	// defined.
```

Comment:
Should we add a comment here that the type names (`BLOB`, `INTEGER PRIMARY KEY`) and so on are case sensitive because of the replacement map?